components/match2/wikis/warcraft/match_group_input_custom.lua

```diff
@@ -78,682 +138,418 @@ function CustomMatchGroupInput._readDate(matchArgs)
-	local template = string.lower(Logic.emptyOr(opponent.template, opponent[1], '')--[[@as string]]):gsub('_', ' ')
+	map.scores = Array.map(opponentInfo, Operator.property('score'))
 
-	if String.isEmpty(template) or template == 'noteam' then
-		opponent = Table.merge(opponent, Opponent.blank(Opponent.team))
-		opponent.name = Opponent.toName(opponent)
-		return opponent
+	if map.finished or MatchGroupInputUtil.isNotPlayed(map.winner, mapInput.finished) then
+		map.resulttype = MatchGroupInputUtil.getResultType(mapInput.winner, mapInput.finished, opponentInfo)
+		map.walkover = MatchGroupInputUtil.getWalkover(map.resulttype, opponentInfo)
+		map.winner = MatchGroupInputUtil.getWinner(map.resulttype, mapInput.winner, opponentInfo)
 	end
 
-	assert(mw.ext.TeamTemplate.teamexists(template), 'Missing team template "' .. template .. '"')
-
-	local templateData = mw.ext.TeamTemplate.raw(template, date)
-
-	opponent.icon = templateData.image
-	opponent.icondark = Logic.emptyOr(templateData.imagedark, templateData.image)
-	opponent.name = templateData.page:gsub(' ', '_')
-	opponent.template = templateData.templatename or template
-
-	return opponent
+	return map, subGroup
 end
 
---MapInput functions
-
----@param match table
----@param mapIndex integer
----@param subGroupIndex integer
----@return integer
-function CustomMatchGroupInput._mapInput(match, mapIndex, subGroupIndex)
-	local map = Json.parseIfString(match['map' .. mapIndex])
-	map.map = (MapsData[(map.map or ''):lower()] or {}).name or map.map
-
-	-- set initial extradata for maps
-	map.extradata = {
-		comment = map.comment or '',
-		header = map.header or '',
-	}
-
-	-- determine score, resulttype, walkover and winner
-	map = CustomMatchGroupInput._mapWinnerProcessing(map)
+---@param mapInput table
+---@param opponentCount integer
+---@return boolean
+function MapFunctions.isFinished(mapInput, opponentCount)
+	local finished = Logic.readBoolOrNil(mapInput.finished)
+	if finished ~= nil then
+		return finished
+	end
 
-	-- get participants data for the map + get map mode + winnerfaction and loserfaction
-	--(w/l faction stuff only for 1v1 maps)
-	CustomMatchGroupInput.ProcessPlayerMapData(map, match, 2)
+	if Logic.isNotEmpty(mapInput.winner) then
+		return true
+	end
 
-	--adjust sumscore for winner opponent
-	if (tonumber(map.winner) or 0) > 0 then
-		match['opponent' .. map.winner].sumscore =
-			match['opponent' .. map.winner].sumscore + 1
+	if Logic.isNotEmpty(mapInput.walkover) then
+		return true
 	end
 
-	-- handle subgroup stuff if team match
-	if match.isTeamMatch then
-		map.subgroup = tonumber(map.subgroup) or (subGroupIndex + 1)
-		subGroupIndex = map.subgroup
+	if Logic.isNotEmpty(mapInput.finished) then
+		return true
 	end
 
-	match['map' .. mapIndex] = map
+	-- check for manual score inputs
+	for opponentIndex = 1, opponentCount do
+		if String.isNotEmpty(mapInput['score' .. opponentIndex]) then
+			return true
+		end
+	end
```

Comment:
Are the scores always like `1-0` / `W-L`? or so
Do we really need this?

Reply:
no they are not always 1-0/W-L
there are RARE cases where they are higher

Reply:
Would that be problematic if it's one that goes up to 3 and scores are entered halfway through?

Reply:
if scores are missing for some opponents that doesn't cause issues

Reply:
Could it be prematurely set to  finished?

Reply:
i have an idea how we potentiall can use commons purely, will need to do some queries and check on it first though

Reply:
techinically just setting `map.bestof=1` will have the exact same effect

Reply:
query result: only used in legacy matches (using legacy wrapper) so finishing them there seems the easier way